src/components/Api.js

```diff
@@ -0,0 +1,193 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+        this._baseUrl = baseUrl;
+        this._headers = headers;
+    }
+    //profile and avatar 
+    async getProfile() {
+        return fetch(this._baseUrl + "/users/me/", 
+        {
+            method: "GET",
+            headers: {
+                authorization: this._headers.authorization,
+            },
+        })
+        .then((res) => {
+            if (res.ok) {
+                return res.json();
+            }
+            console.log("Error");
+            return Promise.reject(`Error : ${res.status}`);
+        }) 
+        .then((result) => {
+            console.log(result);
+            return result;
+        })
+        .catch((err) => console.error(err));
+    }
+
+    async patchProfileAvatar(newLink) {
+      return fetch(this._baseUrl + "/users/me/avatar", {
+        method: "PATCH",
+        headers: {
+          authorization: this._headers.authorization,
+          "Content-Type": "application/json",
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)